soc/arm/nuvoton_numaker/common/pinctrl_soc.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _NUVOTON_NUMAKER_PINCTRL_SOC_H_
+#define _NUVOTON_NUMAKER_PINCTRL_SOC_H_
+
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/types.h>
+
+#define PORT_INDEX(pinmux)    (((pinmux)&0xF0000000) >> 28)
```

Comment:
You could use `FIELD_GET` and `GENMASK` for this pattern.

Reply:
Thanks of your remind. I will consider which one could be  easier mapping v.s data sheet.